language_service/src/signature_help.rs

```diff
@@ -0,0 +1,156 @@
+                    walk_expr(self, args);
+                    if self.signature_help.is_none() {
+                        let callee = unwrap_parens(callee);
+                        if let ast::ExprKind::Path(path) = &*callee.kind {
+                            if let Some(resolve::Res::Item(item_id)) =
+                                self.compilation.unit.ast.names.get(path.id)
+                            {
+                                if let (Some(item), _) = find_item(self.compilation, item_id) {
+                                    if let qsc::hir::ItemKind::Callable(callee) = &item.kind {
+                                        // Check that the callee has parameters to give help for
+                                        if !matches!(&callee.input.kind, hir::PatKind::Tuple(items) if items.is_empty())
+                                        {
+                                            let sig_info = SignatureInformation {
+                                                label: self
+                                                    .display
+                                                    .hir_callable_decl(callee)
+                                                    .to_string(),
+                                                documentation: None,
+                                                parameters: self.get_params(callee),
+                                            };
+
+                                            self.signature_help = Some(SignatureHelp {
+                                                signatures: vec![sig_info],
+                                                active_signature: 0,
+                                                active_parameter: process_args(args, self.offset),
+                                            });
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+                _ => walk_expr(self, expr),
+            }
+        }
+    }
+}
+
+fn unwrap_parens(expr: &ast::Expr) -> &ast::Expr {
+    match &*expr.kind {
+        ast::ExprKind::Paren(inner) => unwrap_parens(inner),
+        _ => expr,
+    }
+}
+
+impl SignatureHelpFinder<'_> {
+    fn get_params(&self, decl: &hir::CallableDecl) -> Vec<ParameterInformation> {
+        let offset = self.display.get_param_offset(decl);
+
+        match &decl.input.kind {
+            hir::PatKind::Discard | hir::PatKind::Bind(_) => {
+                vec![self.make_params_with_offset(offset, &decl.input)]
+            }
+            hir::PatKind::Tuple(items) => {
+                let mut cumulative_offset = offset;
+                items
+                    .iter()
+                    .map(|item| {
+                        let info = self.make_params_with_offset(cumulative_offset, item);
```

Comment:
Extreme nitpicking here, but I feel like this should be called `make_param_with_offset`, singular.

Reply:
done